### PR TITLE
Tests for forget_password view

### PR DESCRIPTION
**Pull request by Byte4Craft**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LVPYTP-44)

## Code reviewers

- [ ] @dmytrorevak
- [ ] @IvanAndrukh
- [ ] @JjuliaP
- [ ] @makarman
- [ ] @noobhunters
- [ ] @Silent777
- [ ] @Staryj812

### Second Level Review

- [ ] @ssOleg
- [ ] @lhalam

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: eventually/tests/unittests/authentication/testforgetpassword.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+VALID_EMAIL_TOKEN = create_token({"user_id":100, 'email': 'john@email.com'}, 60 * 60)
+INVALID_EMAIL_TOKEN = create_token({"user_id":999, 'email': 'john@email.com'}, 60 * 60)
+INVALID_TOKEN = "eyJ1c2VyX2lkIjoxMDAsImVtYWlsIjoiam9obkBlbWFpbC5"
+
+class TestForgetPassword(TestCase):
+    """ Class for Forget Password tests """
+
+    def setUp(self):
+        """ Set up test browser client """
+        self.browser = Client()
```
Comment (lhalam) [inline eventually/tests/unittests/authentication/testforgetpassword.py]: use `self.client`

File: eventually/tests/unittests/authentication/testforgetpassword.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        expect_200 = self.browser.post(link, email_exist, content_type='application/json')
+        self.assertEqual(expect_200.status_code, 200)
+
+    def test_forget_password_put(self):
+        """ Test for forget password PUT function """
+
+        new_password_valid = json.dumps({"new_password": "QwErTy1234"})
+        new_password_invalid = json.dumps({"password": "QwErTy1234"})
+        link_without_token = '/api/v1/user/forget_password/'
+        link_user_exist = link_without_token + str(VALID_EMAIL_TOKEN)[2:-1]
```
Comment (lhalam) [inline eventually/tests/unittests/authentication/testforgetpassword.py]: remove `str`

**Review by lhalam — COMMENTED** [review]
decomposite all tests

File: eventually/tests/unittests/authentication/testforgetpassword.py (hunk 1 above)
Comment (Byte4Craft) [reply eventually/tests/unittests/authentication/testforgetpassword.py]: Fixed, thanks.

File: eventually/tests/unittests/authentication/testforgetpassword.py (hunk 2 above)
Comment (Byte4Craft) [reply eventually/tests/unittests/authentication/testforgetpassword.py]: Token is generated as 'bytes' therefore str + bytes gives an error:
type(EMAIL_TOKEN) : <class 'bytes'>

**Review by skramar — DISMISSED** [review]

File: eventually/tests/unittests/authentication/testforgetpassword.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
+"""
+Forget Password test module
+===========================
+"""
+
+import json
+
```
Comment (redmyt) [inline eventually/tests/unittests/authentication/testforgetpassword.py]: unnecessary blank line

File: eventually/tests/unittests/authentication/testforgetpassword.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+from django.test import TestCase, Client
+
```
Comment (redmyt) [inline eventually/tests/unittests/authentication/testforgetpassword.py]: unnecessary blank line

File: eventually/tests/unittests/authentication/testforgetpassword.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+from authentication.views import CustomUser
+from utils.jwttoken import create_token
+
+VALID_EMAIL_TOKEN = create_token({"user_id":100, 'email': 'john@email.com'}, 60 * 60)
```
Comment (redmyt) [inline eventually/tests/unittests/authentication/testforgetpassword.py]: maybe it is not a good idea to create the token with appropriate custom util method. If that method is wrong the test result would be incorrect.

File: eventually/tests/unittests/authentication/testforgetpassword.py [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+INVALID_EMAIL_TOKEN = create_token({"user_id":999, 'email': 'john@email.com'}, 60 * 60)
+INVALID_TOKEN = "eyJ1c2VyX2lkIjoxMDAsImVtYWlsIjoiam9obkBlbWFpbC5"
```
Comment (redmyt) [inline eventually/tests/unittests/authentication/testforgetpassword.py]: maybe it is better to mock functions that have to create and handle tokens.

File: eventually/tests/unittests/authentication/testforgetpassword.py [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+        user = CustomUser.objects.create(id=100, email="john@email.com")
+        user.set_password = "QwErTy12234"
+        user.save()
+
+
+    def test_post_positive(self):
+        """ Positive test for forget password POST function """
+
+        link = '/api/v1/user/forget_password/'
```
Comment (redmyt) [inline eventually/tests/unittests/authentication/testforgetpassword.py]: maybe it is better to use `url` name in that case, in order to have all test files in one style.

File: eventually/tests/unittests/authentication/testforgetpassword.py [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+        expect_400 = self.client.post(link, {}, content_type='application/json')
+        self.assertEqual(expect_400.status_code, 400)
+
+
+    def test_put_positive(self):
+        """ Positive test for forget password PUT function """
+
+        new_password_valid = json.dumps({"new_password": "QwErTy1234"})
+        link_without_token = '/api/v1/user/forget_password/'
+        link_user_exist = link_without_token + str(VALID_EMAIL_TOKEN)[2:]
```
Comment (redmyt) [inline eventually/tests/unittests/authentication/testforgetpassword.py]: Maybe it is better to mock `handle_token` method and remove this url creating statement.

**Review by redmyt — CHANGES_REQUESTED** [review]

File: eventually/tests/unittests/authentication/testforgetpassword.py (hunk 3 above)
Comment (Byte4Craft) [reply eventually/tests/unittests/authentication/testforgetpassword.py]: Fixed.

File: eventually/tests/unittests/authentication/testforgetpassword.py (hunk 4 above)
Comment (Byte4Craft) [reply eventually/tests/unittests/authentication/testforgetpassword.py]: Fixed.

File: eventually/tests/unittests/authentication/testforgetpassword.py (hunk 5 above)
Comment (Byte4Craft) [reply eventually/tests/unittests/authentication/testforgetpassword.py]: Sorry, forgot about that. Fixed now.

File: eventually/tests/unittests/authentication/testforgetpassword.py (hunk 6 above)
Comment (Byte4Craft) [reply eventually/tests/unittests/authentication/testforgetpassword.py]: There are only 3 constants, guess there's no need to mock functions that create and handle tokens.

File: eventually/tests/unittests/authentication/testforgetpassword.py (hunk 8 above)
Comment (Byte4Craft) [reply eventually/tests/unittests/authentication/testforgetpassword.py]: Moved these duplicated statements to constants

File: eventually/tests/unittests/authentication/testforgetpassword.py (hunk 7 above)
Comment (Byte4Craft) [reply eventually/tests/unittests/authentication/testforgetpassword.py]: Makes sense. Changed names.

**Review by lhalam — APPROVED** [review]

**Review by redmyt — APPROVED** [review]

**Pull request merged**